Gordon360/Services/ProfileService.cs

```diff
@@ -369,6 +369,26 @@ public async Task<FacultyStaffProfileViewModel> UpdateOfficeHoursAsync(string us
             return profile;
         }
 
+        /// <summary>
+        /// mail location setting
+        /// </summary>
+        /// <param name="username"> The username for the user whose mail location is to be updated </param>
+        /// <param name="newMail">The new mail location to update the user's mail location to</param>
+        /// <returns>updated fac/staff profile if found</returns>
+        public async Task<FacultyStaffProfileViewModel> UpdateMailLocationAsync(string username, string newMail)
+        {
+            var profile = GetFacultyStaffProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+            var user = _webSQLContext.accounts.FirstOrDefault(a => a.AD_Username == username);
```

Comment:
```suggestion
            var profile = GetFacultyStaffProfileByUsername(username);
            if (profile == null)
            {
                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
            }
            var user = _webSQLContext.accounts.FirstOrDefault(a => a.AD_Username == username);
```
For performance, I would suggest checking if the `user` `account_type != 'STUDENT'` instead of fetching a faculty/Staff profile. However, this is dependent on whether or not you have access to that column

Reply:
@amos-cha That is how we have been doing for the other updates in fac/staff, would be necessary to change in all those places too?
![image](https://github.com/gordon-cs/gordon-360-api/assets/115328139/96c9a9a2-02d2-469f-bdc6-192538ca9529)
![image](https://github.com/gordon-cs/gordon-360-api/assets/115328139/4090c90a-d19c-4d9f-b0bc-d42e15f2d871)

Reply:
Before that we only have the username we get by AuthUtils.GetUsername(User), so I still think we need these lines